.github/workflows/build.yaml

```diff
@@ -44,3 +45,13 @@ jobs:
 
       - name: Poetry run pytest integration tests
         run: poetry run pytest tests
+
+      - name: Poetry run pytest ci-tests
+        if: "contains(github.event.head_commit.message, '[ci]')"
```

Comment:
@tkilias this is the condition, and I tested it. If you specify "[ci]" keyword in the commit message, ci-tests will be executed, otherwise they are not.

Reply:
Oh did, I miss that. On issue is here, that this skips it, but doesn't fail, when the condition is not true. However, we want to fail if we don't execute the AWS tests, such that we are sure wie executed in the end